File: Problems & Solutions/904. Fruit Into Baskets/07 February 2023/Approach 1 - Brute Force/Fruit Into Baskets.py
class Solution:
    def totalFruit(self, fruits: List[int]) -> int:
        max_picked = 0
        
        for left in range(len(fruits)):
            for right in range(left, len(fruits)):
                basket = set()
                
                for current_index in range(left, right + 1):
                    basket.add(fruits[current_index])
                    
                if len(basket) <= 2:
                    max_picked = max(max_picked, right - left + 1)
        
        return max_picked

File: Problems & Solutions/904. Fruit Into Baskets/07 February 2023/Approach 1 - Brute Force/Fruit Into Baskets.java
class Solution {
    public int totalFruit(int[] fruits) {
        int maxPicked = 0;
        
        for (int left = 0; left < fruits.length; ++left) {
            for (int right = 0; right < fruits.length; ++right) {
                Set<Integer> basket = new HashSet<>();
                
                for (int currentIndex = left; currentIndex <= right; ++currentIndex) {
                    basket.add(fruits[currentIndex]);
                }
                
                if (basket.size() <= 2) {
                    maxPicked = Math.max(maxPicked, right - left + 1);
                }
            }
        }
        
        return maxPicked;
    }
}